colossalai/zero/sharded_model/sharded_model_v2.py

```diff
@@ -216,23 +224,7 @@ def _reduce_scatter_callback(self, param: Parameter, reduced_grad: torch.Tensor)
             # Average grad by world_size for consistency with PyTorch DDP.
             reduced_grad.data.div_(self.gradient_postdivide_factor)
 
-        # Make sure we store fp32 grad
-        reduced_grad.data = cast_tensor_to_fp32(reduced_grad.data)
-
-        # Maybe offload
-        # TODO() optimize GPU->CPU bandwidth utilization
-        if self._cpu_offload:
-            col_move_to_cpu(reduced_grad)
-            # reduced_grad.data = reduced_grad.data.cpu()
-
-        if param.col_attr.grad is None:
-            param.col_attr.grad = reduced_grad.data
-        else:
-            # When dp size = 1
-            # param.col_attr.grad is local accumulated grad shard (full but flatten)
-            # But reduced_grad here is full grad
-            # We should call `view_as`
-            param.col_attr.grad.add_(reduced_grad.data.view_as(param.col_attr.grad))
+        param.col_attr.fp16_grad = reduced_grad.data
```

Comment:
do you need to remove param.grad?

Reply:
Yes. But we cannot set `param.grad` to `None` in parameters' hook. I tried to allocate an empty tensor and then free its storage before returning it, but unit tests failed. I'm debugging this problem.

Reply:
set to None is enough